schema/Readme.md

```diff
@@ -359,207 +364,183 @@ Prisma core provides explicit support for all 3 relation types and connectors mu
   id       Int     @id
-  user     User?
+  user     User    @unique
   address  String
 }
 ```
 
-For 1:1 relationships, it doesn't matter which side you store the foreign key. Therefore Prisma has a convention that the foreign key is added to the model who
-appears first alphanumerically. In the example above, that's the `Customer` model.
-
-Under the hood, the models looks like this:
-
-| **users** |         |
-| --------- | ------- |
-| id        | integer |
-| name      | text    |
+Under the hood, the models looks like this in the database:
 
-| **customers** |         |
-| ------------- | ------- |
-| id            | integer |
-| user          | integer |
-| address       | text    |
+```sql
+create table User (
+  id integer primary key,
+  name text not null
+);
 
-You **may** omit either `User.customer` or `Customer.user` and the relationship will remain intact. This makes either the back-relation or the forward-relation
-optional. If one side of the relation is missing, Prisma implies the field name based on the name of the model it is pointing to.
+create table Customer (
+  id integer primary key,
+  user integer not null references User(id) unique,
+  address text not null
+);
+```
 
-If you're introspecting an existing database and the foreign key does not follow the alphanumeric convention, then we'll use the
-`@relation(_ name: String?, references: Identifier[]?, onDelete: OnDeleteEnum?)` attribute to clarify.
+Here's another valid example where the database backing has been reversed. Additionally, both sides are now optional:
 
 ```groovy
 model User {
   id        Int        @id
-  customer  Customer?  @relation(references: id)
+  customer  Customer?  @unique
   name      String
 }
 
 model Customer {
   id       Int     @id
-  user     User?
+  user     User?   @virtual
   address  String
 }
 ```
 
-##### One-to-Many (1:N) Relationships
-
-A writer can have multiple blogs.
+```sql
+create table User (
+  id integer primary key,
+  customer integer not null references User(id) unique,
+  name text not null
+);
 
-```groovy
-model Writer {
-  id      Int     @id
-  blogs   Blog[]
-}
-
-model Blog {
-  id      Int     @id
-  author  Writer
-}
+create table Customer (
+  id integer primary key,
+  address text not null
+);
 ```
 
-- `Blog.author`: points to the primary key on writer
-
-Connectors for relational databases will implement this as two tables with a foreign-key constraint on the blogs table:
-
-| **writers** |         |
-| ----------- | ------- |
-| id          | integer |
-
-| **blogs** |         |
-| --------- | ------- |
-| id        | integer |
-| author    | integer |
+###### Rules for One-to-One (1:1) Relationships
 
-###### Implied Has-Many
+1. You **must** include both sides of the relation. `User.customer` and `Customer.user`
+2. The virtual back-relation **must be** optional. `User.customer` must be `Customer?`
+3. The forward-relation **must be** `@unique`. `Customer.user` must have `@unique`
```

Comment:
This seems redundant to me. Isn't uniqueness already _implied_ by the fact that I'm modeling this as 1:1?

Reply:
I think @mavilein has a better point on this than me (something like that GraphQL doesn't have any opinion on uniqueness), but I can add that I think this gets back to what type of Schema language do we want to be. 

Using the same example as above:

```sql
create table User (
  id integer primary key,
  customer integer not null references User(id) unique,
  name text not null
);

create table Customer (
  id integer primary key,
  address text not null
);
```

If I removed the `unique` from the example above, I've turned a one-to-one into a one-to-many. The `unique` is what makes it a one-to-one. Each row in the User table must have a unique Customer ID.

Since this PR aims to make everything more explicit, and the unique is required in SQL, so the unique is required here too.

Reply:
I agree with @nikolasburk. In the backend we think that the `@unique` should not be present in the schema in this case. The arity of both fields provides enough information to know that we must add the unique constraint on the database level. Adding this to the PSL does not add any value imo. If we do this we suddenly would have to think about `customer Customer` vs `customer Customer @unique`. These would be two notations with the same meaning.

Reply:
I don't have too much of an opinion on this particular case and would be okay removing the @unique. But I do want to clarify my thinking on this:

My overarching goal is to get to a place where there's no magic and incrementally add magic as we get more data from usage. 

```groovy
model User {
  customer Customer
}

model Customer {
  user User @virtual
}
```

In GraphQL, there's no opinion on the uniqueness in this relationship. User could have many Customers, but just return the first one, or perhaps the last one, or even one Customer at random. It has no opinion there. In my view, the `@unique` says that yes, there's only one of these Customer's per User.

In many ways what I'm suggesting is a step backwards, but it's a step backwards so we can move forward.